lxd/operations/operations.go

```diff
@@ -310,19 +329,28 @@ func (op *Operation) Start() error {
 		return errors.New("Only pending operations can be started")
 	}
 
-	op.status = api.Running
+	err := op.updateStatus(api.Running)
+	if err != nil {
+		op.lock.Unlock()
+		return fmt.Errorf("Failed to update Operation %s status: %w", op.id, err)
+	}
 
 	if op.onRun != nil {
 		go func(op *Operation) {
 			err := op.onRun(op)
 			if err != nil {
+				op.logger.Debug("Failure for operation", logger.Ctx{"err": err})
+
 				op.lock.Lock()
-				op.status = api.Failure
 				op.err = err
+				err = op.updateStatus(api.Failure)
+				if err != nil {
+					op.logger.Warn("Failed to update operation status to failure", logger.Ctx{"err": err})
```

Comment:
Can we include more info in the context of these log messages?


```suggestion
					op.logger.Warn("Failed to update operation status", logger.Ctx{"err": err, "oldStatus": op.status.String(), "newStatus": api.Failure.String()})
```